### PR TITLE
Fixed bugs with updating articles in Apple News, added ability to separately control auto publish and auto update, fixed issue with local time zone not treating articles as being updated

### PR DESCRIPTION
**Pull request by bcampeau**

### CODE REVIEW

File: admin/class-admin-apple-meta-boxes.php [hunk 1]
```diff
@@ -141,9 +141,9 @@ public function publish_meta_box( $post ) {
 		if ( ! empty( $api_id ) ) {
 			$share_url = get_post_meta( $post->ID, 'apple_news_api_share_url', true );
```
Comment (timatron) [inline admin/class-admin-apple-meta-boxes.php]: can you assume this and the meta key below?

Comment (timatron) [inline admin/class-admin-apple-meta-boxes.php]: might just add !empty like $updated_at below for $created_at and $modified_at

File: admin/class-admin-apple-post-sync.php [hunk 2]
```diff
@@ -55,9 +56,15 @@ public function do_publish( $id, $post ) {
...
 		}
 
-		// If the post has been marked as deleted from the API, ignore this update
+		// Proceed based on the current settings for auto publish and update.
+		// Also, if the post has been marked as deleted from the API, ignore this update.
+		$updated = get_post_meta( $id, 'apple_news_api_id', true );
 		$deleted = get_post_meta( $id, 'apple_news_api_deleted', true );
-		if ( $deleted ) {
+		if (
+			( 'yes' != $this->settings->get( 'api_autosync' ) && ! $updated )
```
Comment (timatron) [inline admin/class-admin-apple-post-sync.php]: tabbing looks a bit weird

File: includes/apple-exporter/class-settings.php [hunk 3]
```diff
@@ -19,6 +19,7 @@ class Settings {
 		'api_secret'      => '',
 		'api_channel'     => '',
 		'api_autosync'    => 'yes',
+		'api_autosync_update'    => 'yes',
```
Comment (timatron) [inline includes/apple-exporter/class-settings.php]: pull this empty line

**Comment by timatron** [comment]
few small notes to think through :golf:

File: includes/apple-exporter/class-settings.php (hunk 3 above)
Comment (bcampeau) [inline includes/apple-exporter/class-settings.php]: This is how the original developers formatted this section. I've chosen not to update it to date.

File: admin/class-admin-apple-post-sync.php (hunk 2 above)
Comment (bcampeau) [inline admin/class-admin-apple-post-sync.php]: It's for clarity.

File: admin/class-admin-apple-meta-boxes.php (hunk 1 above)
Comment (bcampeau) [inline admin/class-admin-apple-meta-boxes.php]: This is fine as-is.

**Pull request merged**